packages/instant/src/components/ui/icon.tsx

```diff
@@ -0,0 +1,58 @@
+import * as React from 'react';
+
+import { ColorOption } from '../../style/theme';
+
+type svgRule = 'evenodd' | 'nonzero' | 'inherit';
+interface IconInfo {
+    viewBox: string;
+    fillRule?: svgRule;
+    clipRule?: svgRule;
+    path: string;
+}
+interface IconInfoMapping {
+    failed: IconInfo;
```

Comment:
perhaps we can write it like this?

```
enum IconType {
    Failed,
    Success,
}

interface IconInfoByType {
    [key: IconType]: IconInfo
}
```

Reply:
This suggestion would make it easier to add new icons, I like it

Reply:
Per convo, we would lose some typesafety here, going to leave as-is